docs/userguide.md

```diff
@@ -175,8 +176,53 @@ func MyStep(ctx context.Context, state ewf.State) error {
 - State is saved after each step
 
 ---
+## 9. Queue & Queue Engine
 
-## 9. Testing Workflows
+EWF includes a lightweight, embeddable **queue system** designed for asynchronous, concurrent task execution.
+
+### Queue
+
+The **Queue** is a concurrent-safe structure that allows you to enqueue and process jobs (like workflow runs or background tasks).  
+Each queue automatically starts its worker loop upon creation.
+
+Key features:
+- Thread-safe implementation  
+- Automatic worker pool management  
+- Configurable number of workers  
+- Graceful shutdown via context cancellation  
+
+Example:
+```go
+    client := redis.NewClient(&redis.Options{
+		Addr: "localhost:6379",
+	})
+
+	queue := NewRedisQueue(
+		"queue",
+		"workflow",
+		WorkersDefinition{Count: 1, PollInterval: 200 * time.Millisecond},
+		QueueOptions{AutoDelete: false},
+		client,
+		nil,
+		nil,
+		1*time.Second,
+	)
+	defer queue.Close(context)
```

Comment:
While I understand the reasoning, that doesn't feel ergonomic from an API point of view. Probably we shoudl address [#19](https://github.com/xmonader/ewf/issues/19) as part of the PR  :(